Adapter_Revit_UI/AdapterActions/Pull.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using Autodesk.Revit.UI;
+using BH.oM.Adapter;
+using BH.oM.Adapters.Revit;
+using BH.oM.Data.Requests;
+using System.Collections.Generic;
+
+namespace BH.UI.Revit.Adapter
+{
+    public partial class RevitUIAdapter
+    {
+
+        /***************************************************/
+        /****             Protected Methods             ****/
+        /***************************************************/
+
+        public override IEnumerable<object> Pull(IRequest request, PullType pullType = PullType.AdapterDefault, ActionConfig actionConfig = null)
+        {
+            // Check the document
+            UIDocument uiDocument = this.UIDocument;
+            Document document = this.Document;
+            if (document == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("BHoM objects could not be removed because Revit Document is null.");
```

Comment:
In genera, I think that errors should be formulated add more meaning to what would otherwise be a simple Exception message (`nullReferenceException` in this case, if we just say that something is `null`) - otherwise they might not mean much to the user. Maybe reformulate to say what it means that the Revit Document is null? I personally don't know -- maybe it could be that the connection failed, or something else?

Reply:
Changed to _"BHoM objects could not be removed because Revit Document is null (possibly there is no open documents in Revit)."_